libcxx/test/support/test_iterators.h

```diff
@@ -362,6 +362,111 @@ cpp20_random_access_iterator(It) -> cpp20_random_access_iterator<It>;
 
 static_assert(std::random_access_iterator<cpp20_random_access_iterator<int*>>);
 
+template <std::random_access_iterator It>
```

Comment:
Can you add comment why we need this iterator instead of the normal random-access iterator?

I expect we need to guard this with the proper C++ version.

With these additions and on the `three_way_random_access` I start to wonder whether it makes sense to split these iterators in different files. WDYT?

Reply:
Done.